modules/diskq/diskq-global-metrics.c

```diff
@@ -105,6 +113,205 @@ _update_dir_metrics(gpointer key, gpointer value, gpointer user_data)
   stats_unlock();
 }
 
+static gboolean
+_is_disk_buffer_file(const gchar *filename)
+{
+  const gsize min_len = sizeof("syslog-ng-00000.qf") - 1;
+  const gsize prefix_len = sizeof("syslog-ng-") - 1;
+  const gsize extension_start = sizeof("syslog-ng-00000") - 1;
+
+  gsize len = strlen(filename);
+  if (len < min_len)
+    return FALSE;
+
+  if (strncmp(filename, "syslog-ng-", prefix_len) != 0)
+    return FALSE;
+
+  for (gint i = prefix_len; i < extension_start; i++)
+    {
+      if (filename[i] < '0' || filename[i] > '9')
+        return FALSE;
+    }
+
+  if (strncmp(&filename[extension_start], ".qf", 3) != 0 && strncmp(&filename[extension_start], ".rqf", 4) != 0)
+    return FALSE;
```

Comment:
Sorry, another note because I did not notice:

I think we could be a bit stricter by not allowing any suffixes.
For example, if we don't want to check corrupt files, we should not allow a suffix, otherwise we should look for `corrupted`.

Reply:
Ahh, corrupted buffers are also numbered... Okay... It's not worth all the trouble.

Reply:
Yes :/ I have kept it like this.